pkg/config/types/v1alpha1/generate/generate.go

```diff
@@ -131,98 +169,76 @@ type TrustdInfo struct {
 	Token string
 }
 
-func generateCertificateKey() (string, error) {
-	key, err := randBytes(32)
-	if err != nil {
-		return "", err
+func NewEtcdCA() (ca *x509.CertificateAuthority, err error) {
```

Comment:
I know there is a lot of commonality between the CA funcs, but they are explicit on purpose as a way to contain all related logic in a single function.